### PR TITLE
Cleanup error usage

### PR DESCRIPTION
**Pull request by cpuguy83**
**What this PR does / why we need it**:

Use github.com/pkg/errors for defining errors.
Instead of masking errors with `fmt.Errorf()`, uses `errors.Wrap` where appropriate.
Incidentally this also fixes some formatting issues in certain error messages.

### CODE REVIEW

**Comment by jackfrancis** [comment]
@cpuguy83 thanks for this! In general we like to pave vendor/ changes discretely, and in advance to avoid merge conflicts. But because you authored the PR that induced these conflicts we're probably 👍 here :)

**Comment by cpuguy83** [comment]
Thanks! I did at least put the vendor change in a separate commit here (though I guess I did not on the last change).

This is rebased.

File: cmd/dcos-upgrade.go [hunk 1]
```diff
@@ -76,40 +76,40 @@ func (uc *dcosUpgradeCmd) validate(cmd *cobra.Command) error {
...
 	uc.locale, err = i18n.LoadTranslations()
 	if err != nil {
-		return fmt.Errorf("error loading translation files: %s", err.Error())
+		return errors.Wrap(err, "error loading translation files")
 	}
 
 	if len(uc.resourceGroupName) == 0 {
 		cmd.Usage()
-		return fmt.Errorf("--resource-group must be specified")
+		return errors.New("--resource-group must be specified")
```
Comment (jackfrancis) [inline cmd/dcos-upgrade.go]: Just curious: for error messages that aren't synthesizing downstream `err` objects with add'l string message context, what is the value of using `github.com/pkg/errors`? I.e., what the benefit of `errors.New()` over `fmt.Errorf()`?

Comment (cpuguy83) [reply cmd/dcos-upgrade.go]: `errors.New()` stores the stack information which can aid in debugging... e.g. you can `fmt.Printf("+v")` and see the full stack trace of this error.

**Pull request merged**